File: Modules/ServerTypes/EmployeeGroup/EmployeeGroupShiftPatternForm.ts
import { IntegerEditor, DateEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface EmployeeGroupShiftPatternForm {
    EmployeeRowId: IntegerEditor;
    InsertDate: DateEditor;
    ShiftStartDate: DateEditor;
    ShiftEndDate: DateEditor;
    ShiftId: IntegerEditor;
    UpdateDate: DateEditor;
    DeleteDate: DateEditor;
    IsActive: IntegerEditor;
    InsertUserId: IntegerEditor;
    DeleteUserId: IntegerEditor;
    UpdateUserId: IntegerEditor;
    EmployeeGroupId: IntegerEditor;
}

export class EmployeeGroupShiftPatternForm extends PrefixedContext {
    static formKey = 'EmployeeGroup.EmployeeGroupShiftPattern';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!EmployeeGroupShiftPatternForm.init)  {
            EmployeeGroupShiftPatternForm.init = true;

            var w0 = IntegerEditor;
            var w1 = DateEditor;

            initFormType(EmployeeGroupShiftPatternForm, [
                'EmployeeRowId', w0,
                'InsertDate', w1,
                'ShiftStartDate', w1,
                'ShiftEndDate', w1,
                'ShiftId', w0,
                'UpdateDate', w1,
                'DeleteDate', w1,
                'IsActive', w0,
                'InsertUserId', w0,
                'DeleteUserId', w0,
                'UpdateUserId', w0,
                'EmployeeGroupId', w0
            ]);
        }
    }
}

File: Modules/HumanResource/EmployeeGroup/EmployeeGroupShiftPattern/EmployeeGroupShiftPatternForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeGroup.Forms;

[FormScript("EmployeeGroup.EmployeeGroupShiftPattern")]
[BasedOnRow(typeof(EmployeeGroupShiftPatternRow), CheckNames = true)]
public class EmployeeGroupShiftPatternForm
{
    public int EmployeeRowId { get; set; }
    public DateTime InsertDate { get; set; }
    public DateTime ShiftStartDate { get; set; }
    public DateTime ShiftEndDate { get; set; }
    public int ShiftId { get; set; }
    public DateTime UpdateDate { get; set; }
    public DateTime DeleteDate { get; set; }
    public int IsActive { get; set; }
    public int InsertUserId { get; set; }
    public int DeleteUserId { get; set; }
    public int UpdateUserId { get; set; }
    public int EmployeeGroupId { get; set; }
}